js/core/markdown.js

```diff
@@ -1,280 +1,232 @@
-            ,   HEADERS = /H[1-6]/
-            ;
-
-            function findPosition(header) {
-                return parseInt(header.tagName.charAt(1), 10);
-            }
-
-            function findParent(position) {
-                var parent;
-                while (position > 0) {
-                    position--;
-                    parent = stack[position];
-                    if (parent) return parent;
-                }
-            }
-
-            function findHeader(node) {
-                node = node.firstChild;
-                while (node) {
-                    if (HEADERS.test(node.tagName)) {
-                        return node;
-                    }
-                    node = node.nextSibling;
-                }
-                return null;
-            }
-
-            function addHeader(header) {
-              var section = doc.createElement('section')
-              ,   position = findPosition(header)
-              ;
-
-              section.appendChild(header);
-              findParent(position).appendChild(section);
-              stack[position] = section;
-              stack.length = position + 1;
-              current = section;
-            }
-
-            function addSection(node, process) {
-                var header = findHeader(node)
-                ,   position = header ? findPosition(header) : 1
-                ,   parent = findParent(position)
-                ;
-
-                if (header) {
-                    node.removeChild(header);
-                }
-
-                node.appendChild(process(node));
-
-                if (header) {
-                    node.insertBefore(header, node.firstChild);
-                }
-
-                parent.appendChild(node);
-                current = parent;
-            }
-
-            function addElement(node) {
-                current.appendChild(node);
-            }
+/**
+ * Module core/markdown
+ * Handles the optional markdown processing.
+ *
+ * Markdown support is optional. It is enabled by setting the `format`
+ * property of the configuration object to "markdown."
+ *
+ * We use marked for parsing Markdown:
+ * https://github.com/chjj/marked
+ *
+ * Note that the content of SECTION elements, and elements with a
+ * class name of "note", "issue" or "req" are also parsed.
+ *
+ * The HTML created by the Markdown parser is turned into a nested
+ * structure of SECTION elements, following the structure given by
+ * the headings. For example, the following markup:
+ *
+ *     Title
+ *     -----
+ *
+ *     ### Subtitle ###
+ *
+ *     Here's some text.
+ *
+ *     ### Another subtitle ###
+ *
+ *     More text.
+ *
+ * will be transformed into:
+ *
+ *     <section>
+ *       <h2>Title</h2>
+ *       <section>
+ *         <h3>Subtitle</h3>
+ *         <p>Here's some text.</p>
+ *       </section>
+ *       <section>
+ *         <h3>Another subtitle</h3>
+ *         <p>More text.</p>
+ *       </section>
+ *     </section>
```

Comment:
The `//` to `/* ... */` style change makes it super hard to figure out if the content changed or not.